src/OrderFormAcceptanceTests.Actions/Pages/OrderForm.cs

```diff
@@ -594,12 +594,20 @@ public bool ProposedDateInputIsDisplayed()
 			return Driver.FindElements(Pages.OrderForm.OrderDate).Count == 1;
 		}
 
+		public void EnterProposedDate(DateTime value)
+		{
+			EnterProposedDate(value.Year.ToString(), value.Month.ToString(), value.Day.ToString());
+		}
+
 		public void EnterProposedDate(string year, string month, string day)
 		{
 			Wait.Until(d => d.FindElements(Pages.OrderForm.OrderDateDay).Count == 1);
-			Driver.FindElement(Pages.OrderForm.OrderDateDay).SendKeys(day);
-			Driver.FindElement(Pages.OrderForm.OrderDateMonth).SendKeys(month);
-			Driver.FindElement(Pages.OrderForm.OrderDateYear).SendKeys(year);
+			Driver.FindElement(Pages.OrderForm.OrderDateDay).Clear();
+			Driver.FindElement(Pages.OrderForm.OrderDateMonth).Clear();
+			Driver.FindElement(Pages.OrderForm.OrderDateYear).Clear();
+			Driver.EnterTextViaJs(Wait, Pages.OrderForm.OrderDateDay, day);
+			Driver.EnterTextViaJs(Wait, Pages.OrderForm.OrderDateMonth, month);
+			Driver.EnterTextViaJs(Wait, Pages.OrderForm.OrderDateYear, year);
```

Comment:
Why are we entering the date values via JS?

Reply:
send keys was being flaky. Even when I had a break point and would step through slowly, one field at a time so i know the page is rendered, it was just putting the focus on the field rather than entering the value.